### PR TITLE
Bug 2079958: Add TopologyAwareHints support in kube-proxy.

### PR DESCRIPTION
**Pull request by kyrtapz**
Signed-off-by: Patryk Diak <pdiak@redhat.com>

### CODE REVIEW

File: pkg/network/proxy/hybridproxier.go [hunk 1]
```diff
@@ -115,15 +115,15 @@ func NewHybridProxier(
 }
 
 func (proxier *HybridProxier) OnNodeAdd(node *corev1.Node) {
-	// TODO implement https://github.com/kubernetes/enhancements/pull/640
+	proxier.mainProxy.OnNodeAdd(node)
```
Comment (kyrtapz) [inline pkg/network/proxy/hybridproxier.go:118]: iptables proxiers calls https://github.com/kyrtapz/sdn/blob/f50597184684578f109dc685198f2ad22aa0ec13/vendor/k8s.io/kubernetes/pkg/proxy/iptables/proxier.go#L652, is this an issue?

**Comment by danwinship** [comment]
/lgtm
if you open an origin PR that unskips the skipped test you should be able to get cluster-bot to run the tests against the combined build.

**Comment by kyrtapz** [comment]
For this to work `features.TopologyAwareHints` has to be enabled(It is enabled by default in 1.24).
This means this change will take effect once we rebase sdn to k8s 1.24 which @bpickard22  is working on

/hold

File: pkg/cmd/openshift-sdn-node/kube_proxy.go [hunk 2]
```diff
@@ -2,6 +2,9 @@ package openshift_sdn_node
 
 import (
 	"fmt"
+	"k8s.io/apimachinery/pkg/fields"
```
Comment (martinkennelly) [inline pkg/cmd/openshift-sdn-node/kube_proxy.go:5]: nit: format these

**Comment by kyrtapz** [comment]
/remove-hold
1.24 rebase merged: https://github.com/openshift/sdn/pull/440

**Pull request merged**